### ISSUE TITLE
Hoisted vars in blocks at module scope generate in invalid `d.ts` files when referenced using `typeof`

### ISSUE THREAD

**Issue by lucacasonato** [microsoft/TypeScript#58557]
### 🔎 Search Terms

- hoisting
- var declaration
- declaration files
- typeof
- module

### 🕗 Version & Regression Information

This changed between versions 5.5.0-beta and 5.5.0-dev.20240517

### ⏯ Playground Link

https://www.typescriptlang.org/play/?ts=5.5.0-dev.20240517#code/N4KABBYG4IYE5hmAvGARGg3CAviApgB4AOA9nAC5gA2+VARgFxgUCex+pAZotkA

### 💻 Code

```ts
{
    var a = "";
}
export let b: typeof a;
```

### 🙁 Actual behavior

```ts
export declare let b: typeof a;
```

### 🙂 Expected behavior

d.ts emit of either:

```ts
var a: string;
export declare let b: typeof a;
```

or

```ts
export declare let b: string;
```

Probably it should do the first, then it would be compatible with isolated declarations.

**Comment by lucacasonato** [comment]
@weswigham This is not isolated declarations specific. This is a bug introduced between 5.5.0-beta and 5.5.0-dev.20240517. This reproduction is not related to isolated declaration at all. This code fails to emit correctly / does not raise a diagnostic in the latest dev build, but it does work in 5.5.0-beta.
